hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/filtermanagement/TargetDataExporter.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2020 devolo AG and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.filtermanagement;
+
+import org.eclipse.hawkbit.repository.model.Target;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Utility class to provide formatted Target Data to be used in file exports
+ */
+public class TargetDataExporter {
```

Comment:
In order to implement the above mentioned batch processing you could use the following variant of the exporter:
```suggestion
/**
 * Utility class to export targets of the {@link ProxyTargetFilterQuery} as CSV.
 */
public class TargetDataCsvExporter {
    private static final Logger LOG = LoggerFactory.getLogger(TargetFilterAddUpdateLayout.class);

    private static final List<String> HEADER_KEYS = Arrays.asList("Name", "Description", "Status", "Created By",
            "Created Date", "Modified By", "Modified Date");

    private final String separator;
    private final SimpleDateFormat dateFormat;
    private final TargetManagement targetManagement;

    public TargetDataCsvExporter(final TargetManagement targetManagement, final String separator,
            final String dateFormatPattern) {
        this.targetManagement = targetManagement;
        this.separator = separator;
        this.dateFormat = new SimpleDateFormat(dateFormatPattern);
    }

    public void writeTargetsByFilterIdTo(final Long targetFilterId, final OutputStream out) {
        try {
            addHeader(out);
            convertTargets(targetFilterId, out);

            out.flush();
            out.close();
        } catch (final IOException e) {
            LOG.error("Exception during csv conversion: ", e);
        }
    }

    private void addHeader(final OutputStream out) throws IOException {
        out.write(HEADER_KEYS.stream().collect(Collectors.joining(separator)).concat("\n").getBytes());
    }

    private void convertTargets(final Long targetFilterId, final OutputStream out) throws IOException {
        Pageable pageable = PageRequest.of(0, SPUIDefinitions.PAGE_SIZE);
        Slice<Target> targets;
        do {
            targets = targetManagement.findByTargetFilterQuery(pageable, targetFilterId);
            out.write(targets.stream().map(this::convertTarget).collect(Collectors.joining("\n")).getBytes());
        } while ((pageable = targets.nextPageable()) != Pageable.unpaged());
    }

    private String convertTarget(final Target target) {
        return Arrays
                .asList(target.getControllerId(), target.getDescription(), target.getUpdateStatus().name(),
                        target.getCreatedBy(), toReadableDate(target.getCreatedAt()), target.getLastModifiedBy(),
                        toReadableDate(target.getLastModifiedAt()))
                .stream().map(Strings::nullToEmpty).map(TargetDataCsvExporter::quote)
                .collect(Collectors.joining(separator));
    }

    private String toReadableDate(final long millis) {
        return dateFormat.format(new Date(millis));
    }

    private static String quote(final String input) {
        return String.format("\"%s\"", input);
    }
}
```